src/contexts/price-oracle.context.tsx

```diff
@@ -1,29 +1,47 @@
-import { BigNumber } from "ethers";
+import { formatUnits } from "ethers/lib/utils";
 import { createContext, FC, useContext, useEffect, useState } from "react";
 
 import { useEnvContext } from "src/contexts/env.context";
+import { useGlobalContext } from "src/contexts/global.context";
 import { useProvidersContext } from "src/contexts/providers.context";
+import { Currency, FiatExchangeRates } from "src/domain";
 import { PriceOracle, PriceOracle__factory } from "src/types/contracts/price-oracle";
+import { getFiatExchangeRates } from "src/adapters/fiat-exchange-rates";
+import { USDT_DECIMALS } from "src/constants";
+import * as storage from "src/adapters/local-storage";
 
 interface PriceOracleContext {
-  getTokenPrice: (tokenAddress: string) => Promise<BigNumber>;
+  preferredCurrency?: Currency;
```

Comment:
I'd make this required and default to `USD`.

Reply:
I don't like it, because the source of the default value should be the `localStorage`, and it's already in charge of setting the `defaultValue` to `Currency.USD`. The point is that the `defaultValue` should be synced with `localStorage` always.

Reply:
I understand and you're right. I just try to minimize the optional values that this react context architecture is forcing us to adopt.

And at the end of the day, we have a constant `PREFERRED_CURRENCY` so I see not much conflict on using this value as default also here.

Reply:
Well, at the end it's okay to be `undefined` at the beginning, until we read it from `localStorage` and we take a decision about its current value. I can add the `PREFERRED_CURRENCY` but it's just a way to trick how Context works...